### DISCUSSION TITLE
How to drop properties in bulk

### DISCUSSION THREAD

**Discussion by morpheus2020** [JanusGraph/janusgraph#2650] [Q&A] [answered]
I have a graph backed by Cassandra + ES, which contains 10+ million vertices. These vertices are grouped under 20 or so Vertex labels. I have created a composite index on a property that has the same value as the vertex label. In my case there are ~100k vertices matching the label, I would like to drop a property from these vertices. What would be the best way to do that?

I tried these, however it times out. 
```
v().has('my_indexed_label_prop','mylabel').properties('property_to_remove').drop().iterate()

v().has('my_indexed_label_prop','mylabel').has('property_to_remove').limit(500).properties('property_to_remove').drop().iterate() //in loop
```
Another option I can think of is to use `FulgoraGraphComputer` to get the vertex ids with this property and then drop them in batches.  

What would be my best way to accomplish this?

**Comment by li-boxuan** [comment, marked as answer]
You can use either `FulgoraGraphComputer`, `MapReduce`, or `Spark` to get vertex ids and then drop them in batches. We do similar workload every day by using a `MapReduce` job to scan the whole database and delete applicable ones within the same job.

**Reply by morpheus2020** [reply]
Also is there a way to somehow decode/get the `vertexId` by directly querying ElasticSearch? I

**Reply by li-boxuan** [reply]
JanusGraph stores a string representation of vertexId in the "_id" field of Elasticsearch document. Once you retrieve that "_id", you can use `LongEncoding.decode(_id)` to get the JanusGraph vertex id.

**Reply by morpheus2020** [reply]
This is great! Thank you very much!
